Endava.TechCourse.BankApp.Application/Queries/GetAllReceivedTransfers/GetAllReceivedTransfersHandler.cs

```diff
@@ -12,21 +12,14 @@ public class GetAllReceivedTransfersHandler : IRequestHandler<GetAllReceivedTran
 		public GetAllReceivedTransfersHandler(ApplicationDbContext context)
 		{
 			ArgumentNullException.ThrowIfNull(context);
+
 			_context = context;
 		}
 
 		public async Task<List<Transaction>> Handle(GetAllReceivedTransfersQuery request, CancellationToken cancellationToken)
 		{
-			var receiverWallet =
-				await _context.Wallets.FirstOrDefaultAsync(w => w.OwnerId.ToString() == request.ReceiverId,
-					cancellationToken);
-			if (receiverWallet == null)
-			{
-				throw new ArgumentNullException();
-			}
-
 			var transactions = await _context.Transactions
-				.Where(t => t.ReceiverWalletId == receiverWallet.Id)
+				.Where(t => t.ReceiverId.ToString() == request.ReceiverId)
```

Comment:
If a field is Guid , don't try to transform it to string and check with the one from request. If you have a lot of data in database, it will be faster to parse the string to Guid and to check with all the ids from DB then to transform all Guids to string and check them.